### PR TITLE
[Add Term Entry] CSS Grids: `grid-template-areas`

### PR DESCRIPTION
**Pull request by Nessvah**
### Description

- Closes issue #587 
- Created information about grid-template-areas as well as updated the link from the grid file to match with the path to the grid-template-areas file.
- Added screenshots to examples to hopefully make the information easier to understand.

### Type of Change

- Adding a new entry 

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by Nessvah** [comment]
Thanks for the heads up! That makes a lot of sense. I'll make that change on futures PRs.

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  - 'paths/front-end-engineer-career-path'
+  - 'paths/full-stack-engineer-career-path'
+---
+
+Defines a grid template by referencing the names of the `grid-area` property, establishing cells in the grid and assigning them names.
+The `grid-template` property can be used as a shorthand to define the areas as well as defining the sizes of the columns and rows.
+
+## Syntax
+
+To use `grid-template-areas` it's required to assign names, using `grid-area` property on each grid item. The grid-template-areas are defined in the grid container.
```
Comment (Dusch4593) [inline content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:22]:
```suggestion
Each grid item can be assigned a name with the `grid-area` property. Then, the `grid-template-areas` utilizes these areas `.grid container`.
```

For the "Syntax" section, could we move this text below the code snippet? (also, we should mention that the `.grid-container` must have its `display` property set to `grid`.

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  - 'Layout'
+  - 'Responsive'
+CatalogContent:
+  - 'learn-css'
+  - 'paths/front-end-engineer-career-path'
+  - 'paths/full-stack-engineer-career-path'
+---
+
+Defines a grid template by referencing the names of the `grid-area` property, establishing cells in the grid and assigning them names.
+The `grid-template` property can be used as a shorthand to define the areas as well as defining the sizes of the columns and rows.
```
Comment (Dusch4593) [inline content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:18]:
```suggestion
The `grid-template-areas` property defines a CSS grid by assigning a [`grid-area`](https://www.codecademy.com/resources/docs/css/grids/grid-area) name to each element in the template. The `grid-template` property can be used as a shorthand to define the areas as well as defining the sizes of the columns and rows.
```

I had a question about the last line:

> The `grid-template` property can be used as a shorthand to define the areas as well as defining the sizes of the columns and rows.

Doesn't this information pertain more to an entry about `grid-template` as opposed to one about `grid-template-areas`. (we actually seem to already have [an existing entry](https://www.codecademy.com/resources/docs/css/grids/grid-template) for `grid-template`.

Comment (Dusch4593) [reply content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:18]: Maybe we could write something about how `grid-template-areas` is [collectively used with other properties](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template#constituent_properties) ([`grid-template-columns`](https://www.codecademy.com/resources/docs/css/grids/grid-template-columns) & [`grid-template-rows`](https://www.codecademy.com/resources/docs/css/grids/grid-template-rows)) under the `grid-template` shorthand property?

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  display: grid;
+  max-width: 800px;
+  margin: 0 auto;
+  grid-template-columns: repeat(3, 1fr);
+  grid-template-rows: repeat(4, 1fr 3fr 1fr 1fr);
+  grid-template-areas:
+    ' nav     nav   nav'
+    ' main   main  sidebar'
+    ' main   main  sidebar'
+    'footer footer footer';
```
Comment (Dusch4593) [inline content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:78]:
```suggestion
    'nav    nav      nav'
    'main  main    sidebar'
    'main  main    sidebar'
    'footer footer footer';
```

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+.grid-container {
+  display: grid;
+  max-width: 800px;
+  margin: 0 auto;
+  grid-template-columns: repeat(3, 1fr);
+  grid-template-rows: repeat(4, 1fr 3fr 1fr 1fr);
+  grid-template-areas:
+    ' nav     nav   nav'
+    ' main   main   sidebar'
+    ' main   main   ... '
```
Comment (Dusch4593) [inline content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:115]:
```suggestion
    'nav     nav     nav'
    'main   main   sidebar'
    'main   main   ...';
```

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+.item-three {
+  grid-area: footer;
+}
+
+.grid-container {
+  display: grid;
+  grid-template-areas:
+    'nav nav nav'
+    'main main main'
+    'footer footer footer';
```
Comment (Dusch4593) [inline content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:40]:
```suggestion
    'nav     nav     nav'
    'main   main   main'
    'footer footer footer';
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@Nessvah Finished with my review and I left some comments/suggestions for you. This entry is already looking super solid and I really appreciate you providing the images of the rendered grids!

**Comment by Dusch4593** [comment]
Hey @Nessvah! 👋🏻 Just wanted to check in and see how everything is going on your side of things? I'm happy to help in any way I can 😄

**Comment by Nessvah** [comment]
Hi @Dusch4593 ! 👋🏻 I was just now looking at the doc. I have a question actually. I think you did said that we could let pass the name of the pr branch for now because it could be a little tricky if I did it. But I'm kinda curious and my branch is still with the old name which is incorrect and I really wanna fix it. The command to rename the branch isn't this one: git branch -m <newname> (considering that I'm in the branch that I want to rename. Will that fix the problem once I make a new PR? Or isn't that Simple as it seems? 🙂

**Comment by Dusch4593** [comment]
> I think you did said that we could let pass the name of the pr branch for now because it could be a little tricky if I did it. But I'm kinda curious and my branch is still with the old name which is incorrect and I really wanna fix it. 🙂

Hey @Nessvah 👋🏻 Thank you for the question and double thank you for venturing further into git branching! 🎉 Happy to help! 😄

**TL;DR:**

Unfortunately, it looks like we can't change the name of the remote PR branch when it is attached to an open PR like this one. Deleting it with `git push origin --delete 587` would also delete the comment threads/discussions attached to this PR 😮 Given that reviews are already underway, let's not go through with renaming branches here. However, this is a good practice to be mindful of 😃 and I would totally recommend checking out [this help freeCodeCamp article](https://www.freecodecamp.org/news/how-to-rename-local-and-remote-git-branch/) about renaming both local and remote branches.

> The command to rename the branch is this one: git branch -m (considering that I'm in the branch that I want to rename?? Will that fix the problem once I make a new PR? Or isn't that Simple as it seems?

You're right! The command `git branch -m <old-branch-name> <new-branch-name>` will rename the PR branch on the local end. As a side effect, when changes are pushed with `git push -u origin <new-name>`, the pushed commit(s) prompts GitHub to want to create a new PR altogether.

**Comment by Nessvah** [comment]
@Dusch4593 thank you so much for the patience and time to explain that to me. That makes total sense and now I see why it's best to let just pass this pull request. 😄 in a few hours I'll make the changes to the file and submit the PR to further review. 

I'm glad you could help me through this. Thank you. 👍🏻

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md (hunk 2 above)
Comment (Nessvah) [reply content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:18]: > Maybe we could write something about how `grid-template-areas` is [collectively used with other properties](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template#constituent_properties) ([`grid-template-columns`](https://www.codecademy.com/resources/docs/css/grids/grid-template-columns) & [`grid-template-rows`](https://www.codecademy.com/resources/docs/css/grids/grid-template-rows)) under the `grid-template` shorthand property?

Should I write it on a new example, maybe below the syntax screenshot? So that I can provide another visual explanation of that relationship and how it works.

Comment (Dusch4593) [reply content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:18]: We would just write a sentence here such as the following:

```
This property is collectively used with [`grid-template-columns`](https://www.codecademy.com/resources/docs/css/grids/grid-template-columns) and [`grid-template-rows`](https://www.codecademy.com/resources/docs/css/grids/grid-template-rows) under the [`grid-template`](https://www.codecademy.com/resources/docs/css/grids/grid-template) shorthand property.
```

Comment (Nessvah) [reply content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:18]: Ok thanks! I was about to over explore it. 🫢

**Comment by Nessvah** [comment]
@Dusch4593 Hi Brandon. Please let me know when you can if everything looks good. I was about to make a push when I got an error I tried to fix it and I don't know if it worked.

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  grid-template-rows: repeat(4, 1fr 3fr 1fr 1fr);
+  grid-template-areas:
+    'nav    nav      nav'
+    'main  main    sidebar'
+    'main  main    sidebar'
+    'footer footer footer';
+}
+```
+
+![Example1](https://raw.githubusercontent.com/Codecademy/docs/main/media/grid-template-area2.png)
```
Comment (Dusch4593) [inline content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:81]:
```suggestion
![CSS grid-template-areas Example 1](https://raw.githubusercontent.com/Codecademy/docs/main/media/grid-template-area2.png)
```

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+  grid-template-rows: repeat(4, 1fr 3fr 1fr 1fr);
+  grid-template-areas:
+    'nav     nav     nav'
+    'main   main   sidebar'
+    'main   main   ...'
+    'footer footer footer';
+}
+```
+
+![Example2](https://raw.githubusercontent.com/Codecademy/docs/main/media/grid-template-area3.png)
```
Comment (Dusch4593) [inline content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:119]:
```suggestion
![CSS grid-template-areas Example 2](https://raw.githubusercontent.com/Codecademy/docs/main/media/grid-template-area3.png)
```

Comment (Dusch4593) [reply content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:119]: Can we rename this image stored in `media/`?

~`grid-template-area3.png`~ --> `grid-template-area-example-2.png`

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md (hunk 6 above)
Comment (Dusch4593) [reply content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:81]: Can we rename this image stored in `media/`?

~`grid-template-area2.png`~ --> `grid-template-area-example-1.png`

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+  grid-template-areas:
+    'nav nav nav'
+    'main main main'
+    'footer footer footer';
+}
+```
+
+Each grid item can be assigned a name with the `grid-area` property. Then, the `grid-template-areas` utilizes these areas in the grid container. For this properties to work, `display: grid` must be set in the `.grid-container`.
+
+![Example](https://raw.githubusercontent.com/Codecademy/docs/main/media/grid-template-area1.png)
```
Comment (Dusch4593) [inline content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:43]:
```suggestion
![CSS grid-template-areas Example](https://raw.githubusercontent.com/Codecademy/docs/main/media/grid-template-area1.png)
```

Can we rename this image stored in `media/`?

~`grid-template-area1.png`~ --> `grid-template-area.png`

**Review by Dusch4593 — APPROVED** [review]
@Nessvah Thank you so much for these changes! I left a few last comments/suggestions from me but we should go forward and have the next reviewer take a look at your contribution. 😄

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+```css
+.item-one {
+  grid-area: nav;
+}
+.item-two {
+  grid-area: main;
+}
+.item-three {
+  grid-area: footer;
+}
```
Comment (yangc95) [inline content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:30]: Can we add a new line between each of these here?

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+.grid-container {
+  display: grid;
+  grid-template-areas:
+    'nav     nav    nav'
+    'main   main   main'
+    'footer footer footer';
+}
+```
+
+Each grid item can be assigned a name with the `grid-area` property. Then, the `grid-template-areas` utilizes these areas in the grid container. For this properties to work, `display: grid` must be set in the `.grid-container`.
```
Comment (yangc95) [inline content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:41]: In the last sentence here, can we change "For this properties" -> "For these properties" OR "For this property"

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md (hunk 9 above)
Comment (yangc95) [reply content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:30]: Can we also do this to all successive code blocks? 🙏🏻

File: content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+    'nav     nav     nav'
+    'main   main   sidebar'
+    'main   main   ...'
+    'footer footer footer';
+}
+```
+
+![CSS grid-template-areas Example 2](https://raw.githubusercontent.com/Codecademy/docs/main/media/grid-template-area-example-2.png)
+
+Grid cells can also be left empty with a period `.` instead of white space or names. As long as there is no white space between them, one or more periods can be used for the empty cell.
```
Comment (yangc95) [inline content/css/concepts/grids/terms/grid-template-areas/grid-template-areas.md:121]:
```suggestion
Grid cells can also be left empty with a period `.` instead of white spaces or names. As long as there is no white space between them, one or more period(s) can be used for the empty cell.
```

**Review by yangc95 — COMMENTED** [review]
@Dusch4593 @KTom101 I ended up taking a look at this PR! Thanks @Nessvah for another great submission 🌻

I left only a few points but hope it doesn't conflict with any pending/future suggestions (if any).

Please let me know if there are any questions or concerns!

**Comment by Nessvah** [comment]
Great suggestions! Thanks @yangc95 , will work on it asap. 😊

**Comment by Nessvah** [comment]
Hi guys! @Dusch4593 @yangc95  There's something bothering me because I don't know for sure but... when I'm making chances from the revisions, should I push them and create a new PR and close the old one? I'm trying to make the changes and staying in the same branch and without doing anymore PR but I'm having a hard time doing things that way. So what's the correct way?? Making changes, pulling them to my repo and creating a new PR?

**Comment by Dusch4593** [comment]
It looks like you pushed some changes to your remote `main` branch yesterday, putting it ahead and behind of the the base `Codecademy/docs:main` branch:

Screen Shot 2022-05-11 at 12 46 19 PM

Were these supposed to be pushed to this branch `587`?

**Comment by Nessvah** [comment]
Yes. I thought that they were pushed and were here and then I saw that they are in my repo but don't appear on this pr. I should have taken some notes on the last revision because that happen before and I figure it out. But now I got some doubts about if I was supposed to make a new PR for every time that I work on new revisions and closed the older one.

**Comment by Nessvah** [comment]
I'm gonna try to learn more about contributions and git because doing a contribution is easy and there's a lot of tutorials explaining how but then when there's some revisions and we need to work on those and make some changes, things get a little complicate and other issues arrive and is something I'm still learning. 🙂

**Comment by Dusch4593** [comment]
> I'm gonna try to learn more about contributions and git because doing a contribution is easy and there's a lot of tutorials explaining how but then when there's some revisions and we need to work on those and make some changes, things get a little complicate and other issues arrive and is something I'm still learning. 🙂

I totally agree 😄 I had a lot of trouble with using Git when I first started working with Docs and I still get stumped by it. Learning everyday!

**Comment by Dusch4593** [comment]
> Yes. I thought that they were pushed and were here and then I saw that they are in my repo but don't appear on this pr. I should have taken some notes on the last revision because that happen before and I figure it out. But now I got some doubts about if I was supposed to make a new PR for every time that I work on new revisions and closed the older one.

I've gotten myself in this situation before and haven't yet found a reliable solution other than the following:

1. We can still implement those changes here and finish/merge this PR.

Screen Shot 2022-05-13 at 11 26 29 AM

2. Afterwards, I recommend closing (a) your forked remote repo of `Codecademy/docs` and (b) deleting your existing local `docs` repo by moving to "Trash" and recloning the (new) fork with `git clone`.

For this contribution and your next one, I'm more than to happy to help and share tips on what I do when I work locally/remotely on a PR. 😄

**Comment by Nessvah** [comment]
@Dusch4593 Hi Brandon! I think I got it! The problem is that one time I had an error of fast forward and wasn't really aware of why it happened and how to fix it, so I think the main problem was that I had the remote and the branch where I was working on, out of sync and they had diverged. I really need to keep my repositories in sync and update before doing anything. I would be more than happy if you could share some tips! 😄

**Comment by Nessvah** [comment]
@Dusch4593 @yangc95 Hope that everything looks good now! Let me know if there's more that I can do or if something went wrong.

**Comment by Dusch4593** [comment]
Looks *great* to me, @Nessvah! Thank you so much for your persistence with this PR 😄 🚀 🎉 

@yangc95 How does this PR look for merge? Anything left to add or check?

**Comment by yangc95** [comment]
@Nessvah As a strategy for continuing to improve, I'd suggest creating and documenting a workflow for yourself where you understand what's happening each step of the way! As you're running into issues, it's good to figure out what caused the problem, similar to what you're already doing ☺️

I only caught a glimpse of what you were having trouble with. It looked like you weren't working on the same branch, but it seems like you got the hang of it for this PR! It's important to [merge](https://www.codecademy.com/resources/docs/git/merge) or rebase with the remote repo's main Docs branch into the branch you're working on so we don't hit into divergent branch errors (like above).

My typical workflow is similar to this:

1. I pick up an issue or decide to work on something (typo, new entry)
2. Using git, I go to my forked main branch and `git fetch <docs remote repo>/main`
3. `git merge` or `git rebase` the docs remote repo into my forked main branch. Push up after, and you should see your forked branch is up-to-date on GitHub
4. `git checkout <branch-name>` with a branch name that pertains to what I'm working on
5. `git fetch` and `git merge`/`git rebase` as needed into the branch to keep the branch you're working on, up-to-date with the Docs main repository.
6. `git add` `commit` and `push` the branch, you can follow the git directions here upon the first push and create the PR on GitHub
7. If you work on GitHub after committing our changes/suggestions, make sure to pull any changes before working on your editor+command-line.


Hope this helps! Like @Dusch4593 said, we're learning every day and there's always room for improvement. If you have any my trouble with Git, feel free to create a topic in the Docs forums and maybe we could dive deeper into future issues you run into: https://discuss.codecademy.com/c/community/community-docs/2205 ☺️

**Comment by yangc95** [comment]
> Looks _great_ to me, @Nessvah! Thank you so much for your persistence with this PR 😄 🚀 🎉
> 
> @yangc95 How does this PR look for merge? Anything left to add or check?

I'm going to take one more look later today to make sure everything looks good!

**Review by yangc95 — APPROVED** [review]

**Comment by Dusch4593** [comment]
@Nessvah Congrats and thank you for another great contribution to Docs! 🚀 🎉 🎊 

When the entry goes live, we can see it here: https://www.codecademy.com/resources/docs/css/grids/grid-template-areas

**Comment by Nessvah** [comment]
@Dusch4593 @yangc95 thank you so much for both your help! Christine, that's some amazing tips thanks for sharing. Hope to continue to improve and help building this amazing initiative that docs is. 🙏🏻😊👏🏻
